[12.0][ADD] product_supplierinfo_group

kevinkhao opened the pull request:
Product price values can be tedious and error-prone to enter one by one when you need to repeat the same values:

* vendor,
* product name,
* variant
* product code

Whereas most of the time you just want to enter the minimum quantities, price, dates, delay.

With this module, price lines look to their parent (product.supplierinfo.group) in order to get the values from these repetitive fields.

In addition, it adds a computed field that summarizes all price/quantities deals.

Here are some screenshots for before/after comparison.

Before:
![before_1](https://user-images.githubusercontent.com/19902174/90906163-80fdad00-e3d1-11ea-8aa8-c0f2c2310286.png)
Boring !
![before_2](https://user-images.githubusercontent.com/19902174/90906166-822eda00-e3d1-11ea-94b1-16dfb75e9d3d.png)

After:
![after_1](https://user-images.githubusercontent.com/19902174/90906145-79d69f00-e3d1-11ea-9929-9093192f7ee8.png)
Much better
![after_2](https://user-images.githubusercontent.com/19902174/90906157-7e9b5300-e3d1-11ea-9bae-980d6d49f5c3.png)

Commit:
[ADD] product_sequence: Allows to associate a sequence to the product reference. The reference (default code) is unique (SQL constraint) and required.

sebastienbeau committed:
[MERGE] merge translation commited by launchpad on extra-addons6.0

bguillot committed:
[FIX] fix write all product with default_code together and then write each product without default_code

eLBati committed:
[FIX] resetting sequence code when updating module

Commit:
[REM] use Model instead of osv.osv

some minor reformatting in the process

sebastienbeau committed:
[FIX] product_sequence : write should support list or integer

Commit:
[MIGR] migration of product_sequence to openerp 7.0 (required for product_images)

Commit:
FIX description of module product_custom_attributes.
Version numbers should only have 2 digits.

Commit:
Launchpad automatic translations update.

Commit:
Set all addons to uninstallable and move them into __unported__ folder

Commit:
MIG product_sequence

Commit:
Added german translation

yvaucher committed:
Add bug tracker link on README.rst

gdgellatly committed:
product_sequence

Add pre_init_hook to manually set nulls and '/' default codes
 to a unique code to greenify branch
add pragma to pre_init method as it must have completed successfully
if module installs

yvaucher committed:
Add missing default oca icons

atchuthan committed:
[PORT][9.0] product_m2m_categories/product_multi_category

SodexisTeam committed:
product_sequence (#156)

atchuthan committed:
[MIG][10.0] migrate product_sequence

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[9.0][ADD] product_pricelist_direct_print (#223)

angelmoya committed:
MIG 10.0 product_multi_category

FIX product_multi_category README.rst

oca-transbot committed:
OCA Transbot updated translations from Transifex

damcar committed:
[MIG] product_multi_category: Migration to 11.0

carlosdauden committed:
[IMP] product_pricelist_direct_print: Add order option + change widget for pricelist (#290)

agyamuta committed:
[ADD] Module to add a constraint on the internal reference of the product to make it unique across the database.

LoisRForgeFlow committed:
[11.0][MIG/IMP] product_sequence:
* Migration to v11
* Support sequence by Product Category

JordiBForgeFlow committed:
fixes and adds tests

agyamuta committed:
[11.0] product_internal_reference_mandatory (#334)

* [ADD] Module to set the  internal reference of the product as required.

sbidoul committed:
remove obsolete .pot files [ci skip]

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update product_code_unique.pot

oca-travis committed:
[UPD] Update product_code_mandatory.pot

oca-travis committed:
[UPD] Update product_multi_category.pot

oca-travis committed:
[UPD] Update product_sequence.pot

chienandalu committed:
[MIG] product_pricelist_direct_print: Migration to 11.0

sergio-teruel committed:
[11.0][NEW] product_secondary_unit: New module for setting a secondary unit for a product (#377)

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update product_secondary_unit.pot

ernestotejeda committed:
product_pricelist_direct_print: extend module

Print price list for future date
Print price list for N last ordered products
Send report by email to each partner selected

sergio-teruel committed:
[11.0][ADD] product_secondary_unit: Add default secondary unit for purchases

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (17 of 17 strings)

Translation: product-attribute-11.0/product-attribute-11.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-11-0/product-attribute-11-0-product_secondary_unit/fr/

sergio-teruel committed:
[11.0][ADD] product_cost_security: New module to restrict product cost view

sergio-teruel committed:
[11.0][ADD] product_cost_security: Fix translate folder name

sergio-teruel committed:
[11.0][FIX] product_pricelist_direct_print: Remove duplicate default code

bizzappdev committed:
[MIG]product_code_unique: migration from 11 to 12

sudhir-erpharbor committed:
[MIG] Migrated product_code_mandatory to v12

alexey-pelykh committed:
[MIG] product_pricelist_direct_print: Migration to 12.0

sudhir-erpharbor committed:
[MIG] Migrated product_multi_category to v12

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

oca-travis committed:
[UPD] Update product_secondary_unit.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

sudhir-erpharbor committed:
[MIG] Migrated product_sequence to v12

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-11.0/product-attribute-11.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-11-0/product-attribute-11-0-product_secondary_unit/es/

sudhir-erpharbor committed:
[ADD] Added Readme.

sergio-teruel committed:
[11.0][IMP] product_cost_security: Avoid use odoo PR 28755
https://github.com/odoo/odoo/pull/28755

sergio-teruel committed:
[11.0][IMP] product_secondary_unit: Make secondary unit name translatable

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update product_multi_category.pot

oca-travis committed:
[UPD] Update product_cost_security.pot

kittiu committed:
[12.0][MIG] product_secondary_unit

oca-travis committed:
[UPD] Update product_secondary_unit.pot

patrickrwilson committed:
[MIG][12.0] product_template_tags

[UPD] Suggested Changes

Updated code based on reviews

[UPD] Add readme folder

Added readme folder and made it so tags can be colored.

[UPD] Aesthetic Changes

Aesthetic changes requested from reviewer.

[UPD] README.rst

[UPD] Update product_template_tags.pot

max3903 committed:
[DEL] Cleanup pyc files

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_multi_category
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_multi_category/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_secondary_unit/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_template_tags/

[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[IMP] product_sequence: Suggested PR changes

rven committed:
[MIG] product_code_unique: Migration to 13.0

oca-travis committed:
[UPD] Update product_sequence.pot

oca-travis committed:
[UPD] Update product_code_unique.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
pre-commit, black, isort

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_sequence/

digitalsatori committed:
Added translation using Weblate (Chinese (Simplified))

digitalsatori committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-12.0/product-attribute-12.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_secondary_unit/zh_CN/

zeroheure committed:
typo

OCA-git-bot committed:
[UPD] README.rst

zeroheure committed:
FIX remove README instrucions

[UPD] README.rst

sergio-teruel committed:
[MIG] product_cost_security: Migration to v12.0

OCA-git-bot committed:
product_template_tags 12.0.1.1.0

oca-travis committed:
[UPD] Update product_cost_security.pot

ps-tubtim committed:
[IMP] product_secondary_unit: black, isort

ps-tubtim committed:
[MIG] product_secondary_unit: Migration to 13.0

sergio-teruel committed:
[12.0][FIX] product_sequence: Keep default_code when copy products if dictionary values contains the default_code key.

ps-tubtim committed:
[IMP] product_multi_category: black, isort

OCA-git-bot committed:
product_sequence 12.0.1.0.1

ps-tubtim committed:
[MIG] product_multi_category: Migration to 13.0

ps-tubtim committed:
[IMP] product_template_tags: black, isort

ps-tubtim committed:
[MIG] product_template_tags: Migration to 13.0

[UPD] Update product_template_tags.pot

oca-travis committed:
[UPD] Update product_secondary_unit.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_cost_security
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_cost_security/

sergio-teruel committed:
[IMP] product_sequence: black, isort

sergio-teruel committed:
[MIG] product_sequence: Migration to v13.0

oca-travis committed:
[UPD] Update product_sequence.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-13.0/product-attribute-13.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_sequence/

sergio-teruel committed:
[IMP] product_cost_security: black, isort

sergio-teruel committed:
[MIG] product_cost_security: Migration to v13.0

oca-travis committed:
[UPD] Update product_cost_security.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-13.0/product-attribute-13.0-product_cost_security
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_cost_security/

sergio-teruel committed:
[IMP] product_pricelist_direct_print: black, isort

sergio-teruel committed:
[MIG] product_pricelist_direct_print: Migration to v13.0

sergio-teruel committed:
[MIG] product_supplierinfo_for_customer: Migration to v13.0

emagdalenaC2i committed:
[ADD] project_revision

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-13.0/product-attribute-13.0-product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_template_tags/

benbrich committed:
Translated using Weblate (German)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_sequence/de/

HviorForgeFlow committed:
[13.0][FIXUP] .isort.cfg

oca-travis committed:
[UPD] Update product_code_mandatory.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update product_bom_revision.pot

OCA-git-bot committed:
[UPD] README.rst

emagdalenaC2i committed:
[FIX] Readme duplicate bug tracker

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_bom_revision 12.0.1.0.1

sebastienbeau committed:
remove duplicate bug tracker

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] product_secondary_unit: Remove product secondary units if a product is deleted

alan196 committed:
Added translation using Weblate (Spanish (Mexico))

alan196 committed:
Translated using Weblate (Spanish (Mexico))

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_secondary_unit/es_MX/

OCA-git-bot committed:
product_secondary_unit 13.0.1.1.0

oca-travis committed:
[UPD] Update product_multi_category.pot

OCA-git-bot committed:
pre-commit update

OCA-git-bot committed:
pre-commit update

OCA-git-bot committed:
pre-commit update

OCA-git-bot committed:
pre-commit update

OCA-git-bot committed:
pre-commit update

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-13.0/product-attribute-13.0-product_multi_category
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_multi_category/

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (3 of 3 strings)

Translation: product-attribute-12.0/product-attribute-12.0-product_code_mandatory
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_code_mandatory/pt/

EmailsGmails committed:
[IMP] product_dimension: black, isort, prettier

Tatider committed:
[MIG] product_dimension: Migration To 13.0

[UPD] Update product_dimension.pot

[UPD] README.rst

kikopeiro committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_template_tags/es/

emagdalenaC2i committed:
[IMP] product_bom_revision: black, isort, prettier

sergio-teruel committed:
[IMP] product_cost_security: Add user_root to security group

OCA-git-bot committed:
product_cost_security 13.0.1.1.0

oca-travis committed:
[UPD] Update product_bom_revision.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[IMP] product_code_mandatory: black, isort, prettier

Commit:
[13.0][MIG] product_code_mandatory

oca-travis committed:
[UPD] Update product_code_mandatory.pot

OCA-git-bot committed:
[UPD] README.rst

didierdonze committed:
Translated using Weblate (German)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_multi_category
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_multi_category/de/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_multi_category
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_multi_category/fr/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 100.0% (11 of 11 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_dimension/fr/

didierdonze committed:
Translated using Weblate (German)

Currently translated at 100.0% (11 of 11 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_dimension/de/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_secondary_unit/fr/

didierdonze committed:
Translated using Weblate (German)

Currently translated at 100.0% (17 of 17 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_template_tags/de/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 100.0% (17 of 17 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_template_tags/fr/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_sequence/fr/

Tonow-c2c committed:
[IMP][13.0] Add product_template_tags_code module

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-13.0/product-attribute-13.0-product_multi_category
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_multi_category/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-13.0/product-attribute-13.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_dimension/

kevinkhao committed:
[12.0][ADD] product_supplierinfo_group

sebastienbeau commented:
@legalsylvain maybe this PR can interest you ;)

bealdav commented:
Ergonomy ideas: 

1/ make it compatible with this module https://github.com/OCA/web/tree/12.0/web_tree_duplicate#configuration : set duplicate=True
2/ Use double column per row for fields in supplier info group

To discuss

kevinkhao committed:
[IMP] Separate rows and add nolabel on supplierinfo group view

kevinkhao commented:
> Ergonomy ideas:
> 
> 1/ make it compatible with this module https://github.com/OCA/web/tree/12.0/web_tree_duplicate#configuration : set duplicate=True
> 2/ Use double column per row for fields in supplier info group
> 
> To discuss

Thanks, view change is applied

sebastienbeau commented on product_supplierinfo_group/models/product_supplierinfo_group.py:
```diff
@@ -0,0 +1,66 @@
+        help="If not set, the vendor price will apply to all "
+        "variants of this product.",
+    )
+    name = fields.Many2one(
```
I know that odoo use "name" for the link to the partner but as it's an new object I prefers to use "partner_id" here.

rousseldenis replied:
Agree

kevinkhao replied:
done

kevinkhao committed:
[IMP] rename "name" to "partner_id" on product_supplierinfo_group

rousseldenis commented on product_supplierinfo_group/models/product_supplierinfo.py:
```diff
@@ -0,0 +1,69 @@
+class ProductSupplierinfo(models.Model):
+    _inherit = "product.supplierinfo"
+
+    supplierinfo_group_id = fields.Many2one("product.supplierinfo.group", required=True)
```
I think you maybe needs to set a default (see runbot warning)

kevinkhao replied:
@rousseldenis thanks for the remark
This meant a bit more head scratching than expected :)
see: https://github.com/OCA/product-attribute/pull/665/commits/46a04187fbd6595c6db4626eb2ef0a4ad0561f94

kevinkhao committed:
[FIX] Allow for smooth install of module due to new required=True field on supplierinfo

sebastienbeau commented on product_supplierinfo_group/views/product.xml:
```diff
@@ -0,0 +1,21 @@
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <field name="seller_ids" position="replace">
```
Hi, you should avoid using replace as it can break other module set invisible instead

Moreover there is two field seller_ids and variant_seller_ids we should hide both see here

https://github.com/OCA/OCB/blob/12.0/addons/purchase/views/product_views.xml#L14

kevinkhao committed:
[IMP] Related fields to store=True for better stability when uninstalling this module
Appearance overhaul
Cleanup

kevinkhao committed:
[FIX] Small fix on name field

nikul-serpentcs commented on product_supplierinfo_group/models/product_supplierinfo.py:
```diff
@@ -0,0 +1,74 @@
+        related="supplierinfo_group_id.product_tmpl_id", store=True
+    )
+    name = fields.Many2one(
+        related="supplierinfo_group_id.partner_id", required=False, store=True
```
@kevinkhao Here no need to add `required=False`, Because by default `required False`.

kevinkhao replied:
@nikul-serpentcs thanks for the catch

kevinkhao replied:
actually we do need to set it to required=False https://github.com/odoo/odoo/blob/d87111ae3e3cf3d654180c986745623ebe731667/addons/product/models/product.py#L743 if we don't then we have problems when trying to create a new record

nikul-serpentcs commented on product_supplierinfo_group/models/product_supplierinfo_group.py:
```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2015-Today GRAP (http://www.grap.coop)
+# Copyright (C) 2020-Today Akretion (http://www.akretion.com)
+# @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License GPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
```suggestion
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

kevinkhao replied:
done

kevinkhao committed:
[IMP] Revert useless name=required argument

kevinkhao committed:
[IMP] GPL->AGPL, exclude incompatible module product_supplierinfo_for_customer

kevinkhao committed:
[FIX] re-add required=False attribute

sebastienbeau committed:
[FIX] fix issue when installing on empty db

Skip the hook if there is no data to migrate
and remove useless commit

sebastienbeau commented:
@kevinkhao I had pushed a fix when installing on an empty db.
Also never ever use cr.commit() do it only if you have a really really good reason do to it (really really rare case maybe 1 time every 10 years)

kevinkhao commented:
@sebastienbeau noted with thanks
it still gets stuck on the runbot, adding a temporary trivial commit to regenerate logs (they are currently inaccessible)
I also just realized I didn't yet do a PR for the product_supplierinfo_group_margin yet because of the red status

kevinkhao committed:
[TMP] Trivial commit for runbot reset

sebastienbeau reviewed (approved):
Code review and test.
Everything is green now

bealdav reviewed (changes requested):
Thanks a lot for this module. Just tested, it works very well.
Just a point

You should define a domain for product_id because here it's possible to mix product_id with a product.template which don't have a matching product_tmpl_id.

![Selection_080](https://user-images.githubusercontent.com/1853434/93617617-7df8db00-f9d6-11ea-9e78-7b46a72c0e32.png)
bolt != desk

kevinkhao commented:
> Thanks a lot for this module. Just tested, it works very well.
> Just a point
> 
> You should define a domain for product_id because here it's possible to mix product_id with a product.template which don't have a matching product_tmpl_id.
> 
> ![Selection_080](https://user-images.githubusercontent.com/1853434/93617617-7df8db00-f9d6-11ea-9e78-7b46a72c0e32.png)
> bolt != desk

Thank you for the review.

What you are saying makes sense. In the last commit I tried to do this restriction. It works great on product.product, but not on product.template. I am suspecting it is a limitation of the JS side of Odoo, more specifically the popup of when you click the One2many widget on "Add a line..."

Something funky happens when you reference the parent. In our case it is product_tmpl_id on product.supplierinfo.group. Because product_tmpl_id is on our popup and refers to the active_id of the screen that is behind the popup. Then the onchange seems broken.

From the product template screen:

![Screenshot from 2020-09-23 21-41-37](https://user-images.githubusercontent.com/19902174/94061518-c99bf200-fde5-11ea-939f-bef5d76a3345.png)

Nothing happens.

From the product.product screen:

![Screenshot from 2020-09-23 21-41-51](https://user-images.githubusercontent.com/19902174/94061582-dd475880-fde5-11ea-9952-9d00727daad8.png)

Works great.


I will need to spend a bit more time thinking about it. Do you have some idea ?

kevinkhao committed:
[IMP] Add domain on product_id for supplierinfo.group object

kevinkhao commented:
Fixed the issue.

The problem was due to the fact that in a One2many field you need to specify in context the default value of the parent (here the child is supplierinfo.group and the parent is product_tmpl_id) if you want to use it in domains on onchanges. Or else, it will only be set on write (so, only when you click "Save", before that you won't have access to it for onchanges/domains...).

bealdav reviewed (approved):
LGTM, thanks fot the fix

moylop260 committed:
[IMP] product_template_tags: speed up tag.products_count compute

The tag.products_count field was store=True
If more than one user is adding/removing the same tag to different products
at the same time it will raise concurrencies issues

In fact, if you are configuring/importing your products
this compute could be so slow
Since that each tag will be updated for each product

It was changed to store=False in order to avoid these issues.

The tag.products_count field is displayed in tree view
So it needs to be so faster
Then the compute was changed using a query directly with "group by"
in order to process it faster

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

OCA-git-bot committed:
[UPD] README.rst

emagdalenaC2i committed:
[IMP] product_dimension: black, isort, prettier

emagdalenaC2i committed:
[MIG] product_dimension: Migration to 14.0

[UPD] Update product_dimension.pot

[UPD] addons table in README.md

[UPD] README.rst

OCA-git-bot committed:
product_template_tags 13.0.2.0.0

OCA-git-bot committed:
[ADD] setup.py

Commit:
[REF] run precommit

Commit:
[FIX] _compute_products_count
avoid return if nothing to compute to be compliant with new ORM compute field method

Commit:
[IMP] add unicity constraint on name and company_id

emagdalenaC2i committed:
[IMP] product_bom_revision: black, isort, prettier

emagdalenaC2i committed:
[MIG] product_bom_revision: Migration to 14.0

alexey-pelykh committed:
[UPD] Brainbean Apps => CorporateHub

watthanun committed:
[IMP] product_code_unique: black, isort, prettier

watthanun committed:
[MIG] product_code_unique: Migration to 14.0

watthanun committed:
[IMP] product_code_mandatory : black, isort, prettier

alan196 committed:
[IMP] product_secondary_unit: black, isort, prettier

alan196 committed:
[MIG] product_secondary_unit: Migration to 14.0

watthanun committed:
[MIG] product_cost_security: Migration to 14.0

watthanun committed:
[MIG] product_multi_category: Migration to 14.0

OCA-git-bot committed:
Merge PR #702 into 14.0

Signed-off-by sebastienbeau

oca-travis committed:
[UPD] Update product_template_tags.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
Merge PR #717 into 14.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update product_secondary_unit.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
Merge PR #713 into 14.0

Signed-off-by dreispt

oca-travis committed:
[UPD] Update product_code_unique.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

watthanun committed:
[MIG] product_sequence: Migration to 14.0

Commit:
[ADD] product_order_noname: Speedup product retrieve

This module improves retrieving products when a language is installed
and configured. The default behaviour is to order products by name,
which in this case would be slow because Odoo will process the
product's name to translate it.

When ordering products by name and having a language configured the
Odoo's ORM makes a join to `product_template` an then another join but
this time with a nested SELECT to retrieve the translation. This is very
slow because the lack of index on the JOIN, making a query like
`env['product.product'].search([], limit=10)` taking up to 1027.282 ms.

With the changes introduced here the query is considerably simpler, just
as the following:

    SELECT "product_product".id
    FROM   "product_product"
    WHERE  ( "product_product"."active" = true )
    ORDER  BY "product_product"."default_code"
    LIMIT  10

Also the execution time dramatically decrease, from 1027.282 ms to
0.624 ms which is about 1.65k times faster.

OCA-git-bot committed:
Merge PR #704 into 14.0

Signed-off-by rousseldenis

oca-travis committed:
[UPD] Update product_bom_revision.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
product_bom_revision 14.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
Merge PR #721 into 14.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update product_cost_security.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
Merge PR #727 into 14.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update product_sequence.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
Merge PR #723 into 14.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update product_multi_category.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

watthanun committed:
[MIG] product_code_mandatory: Migration to 14.0

joao-p-marques committed:
[IMP] product_pricelist_direct_print: black, isort, prettier

joao-p-marques committed:
[MIG] product_pricelist_direct_print: Migration to 14.0

TT26448

moylop260 committed:
[REF] product_order_noname: Change product.template order

Product.template has `_order='name'`
even if the `name` field has an `index=True`
This index is not used because of this field has a
`translate=True` parameter

So, it is in the same situation like product.product

So, it is changed using `_order='default_code, id'` instead.
For a faster sort it requires an index so the field `default_code`
is redefined to `index=True`

OCA-git-bot committed:
product_order_noname 12.0.2.0.0

moylop260 committed:
[FIX] pre-commit: Requested docutils<0.15.1.1,>=0.15.1 from ... has different version in metadata: '0.15.1' just bumping version of oca-mt repo

OCA-git-bot committed:
Merge PR #739 into 14.0

Signed-off-by moylop260

oca-travis committed:
[UPD] Update product_order_noname.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
Merge PR #719 into 14.0

Signed-off-by rousseldenis

oca-travis committed:
[UPD] Update product_code_mandatory.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_code_mandatory 14.0.1.0.1

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[IMP] update dotfiles

OCA-git-bot committed:
Merge PR #742 into 14.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update product_pricelist_direct_print.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

sbidoul committed:
Resolve pre-commit .rej update conflict

OCA-git-bot committed:
Merge PR #751 into 14.0

Signed-off-by sbidoul

sebastienbeau commented:
/ocabot merge nobump
